tests/integration/utils.py

```diff
@@ -104,41 +111,69 @@ def make_request(self, endpoint, expected_status_code,
 
 # Check the schema of response match OpenAPI specification
 def check_schema(self, response, schema):
+    # Mapping of OpenAPI data types and python data types
+    types_dict = {
+        'string': str,
+        'number': float,
+        'integer': int,
+        'int32': int,
+        'int64': int,
+        'float': float,
+        'double': float,
+        'boolean': bool,
+        'array': list,
+        'object': dict
+    }
+
     # Helper function to get attributes of the schema
     def __get_schema_attributes():
         return schema['attributes']['properties']
 
     # Helper function to map between OpenAPI data types and python data types
     def __get_attribute_type(attribute):
-        types_dict = {
-            'string': str,
-            'integer': int,
-            'int32': int,
-            'int64': int,
-            'float': float,
-            'double': float,
-            'boolean': bool,
-            'array': list,
-            'object': dict
-        }
-
+        openapi_type = None
         if 'properties' in attribute:
             return dict
         elif 'format' in attribute and attribute['format'] in types_dict:
             openapi_type = attribute['format']
         elif 'type' in attribute:
             openapi_type = attribute['type']
-        else:
-            logging.warning('OpenAPI property contains no type or properties')
-            return None
+        elif '$ref' in attribute:
+            openapi_type = __get_object_type(attribute['$ref'])
+
+        if openapi_type:
+            return types_dict[openapi_type]
```

Comment:
Line 134 is not really necessary and you can shorten this line to be something like:

```python
return types_dict[openapi_type] if openapi_type else None
```